conans/client/conanfile/configure.py

```diff
@@ -10,8 +10,9 @@ def run_configure_method(conanfile, down_options, profile_options, ref):
     if hasattr(conanfile, "config_options"):
         with conanfile_exception_formatter(conanfile, "config_options"):
             conanfile.config_options()
-    else:
-        default_config_options(conanfile)
+    elif conanfile.implements:
```

Comment:
```suggestion
    elif "auto_shared_fpic" in conanfile.implements:
         auto_shared_fpic_config_options(conanfile)
```
I think I'd have preferred a minimal, explicit implementation to start with this feature, and only generalize a bit later when there is more evidence.

Reply:
(But I am fine with it, not a big issue)

Reply:
Yes, I did it this way in case we wanted to respect the order of execution stated in the list, in fact, I prefer the minimal, explicit implementation

Reply:
I'd say lets hardcode the order, we know it works. If in the future order-defined application of ``implements`` is really a thing... maybe they should be explicit in their methods.

Yeah, I think we shouldn't care about the user order at all, it would be another point of variability and issues.